### ISSUE TITLE
Recognise cursor column for completion result

### ISSUE THREAD

**Issue by christian-bromann** [arshad-yaseen/monacopilot#81]
First of, THANK YOU 🙏  this project is amazing!

- `monacopilot` version: 0.15.0
- `monaco-editor` version: 0.52.0
- `node` version: 22.11.0
- `pnpm` version: 9.4.0

Relevant code or config

```javascript
const copilot = new Copilot(process.env.ANTHROPIC_API_KEY!, {
  provider: 'anthropic',
  model: 'claude-3-5-sonnet',
})
const completion = await copilot.complete({ body })
```

What you did: everything according to the docs.

What happened: The completion metadata looks as following:

```
{
   completionMetadata: {
     language: 'json',
     textBeforeCursor: '{\n' +
       '  "tags": [\n' +
       '    "Contacts"\n' +
       '  ],\n' +
       '  "summary": "Get Contact List",\n' +
       '  "parameters": [\n' +
       '    {\n' +
       '      "name": "Authorization",\n' +
       '      "in": "header",\n' +
       '      "schema": {\n' +
       '        "type": "string",\n' +
       '        "description": "Bearer authentication token"\n' +
       '      },\n' +
       '      "example": "Bearer {{token}}"\n' +
       '    }\n' +
       '  ],\n' +
       '  "responses": {\n' +
       '    "200": {\n' +
       '      "description": "OK",\n' +
       '      "content": {\n' +
       '        "text/plain": {\n' +
       '          "schema": {\n' +
       '            "type": "array",\n' +
       '            "items": {\n' +
       '              "type": "object",\n' +
       '              "pro',
```

Problem description: the cursor may not be at the beginning of the editor, however the completion result returns a result that assumes that.

Suggested solution: as a workaround I currently have the following code:

```ts
  const initialSpaces = body.completionMetadata.textBeforeCursor.split('\n').pop()?.match(/^\s+/)?.[0] || ''
  if (initialSpaces.length > 0 && completion.completion) {
    completion.completion = completion.completion.replace(/\n/g, `\n${initialSpaces}`)
  }
```

I am sure this may be useful for anyone using the package so I wonder if it makes sense to make this the default behavior.

Thanks again!

**Comment by arshad-yaseen** [comment]
Thank you for pointing out this, Will fix and let you know!

**Comment by arshad-yaseen** [comment]
Could you please share a screen record for clarity if possible?

**Comment by arshad-yaseen** [comment]
@christian-bromann Fixed it in `v0.15.2` 🚀

**Comment by christian-bromann** [comment]
Awesome, thanks a ton!
